pkg/errors/withstatus.go

```diff
@@ -0,0 +1,58 @@
+package errors
+
+import (
+	"errors"
+
+	"google.golang.org/genproto/googleapis/rpc/errdetails"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+	"google.golang.org/protobuf/runtime/protoiface"
+
+	v1 "github.com/authzed/authzed-go/proto/authzed/api/v1"
+)
+
+// Domain is the domain used for all errors.
+const Domain = "authzed.com"
+
+// WithCodeAndDetails returns a gRPC status message containing the error's message, the given
+// status code and any supplied details.
+func WithCodeAndDetails(err error, code codes.Code, details ...protoiface.MessageV1) *status.Status {
+	status, err := status.New(code, err.Error()).WithDetails(details...)
+	if err != nil {
+		panic(err)
```

Comment:
I'd give an explanation as to why we're panicking here.

I looked into the code and saw it's if you pass an OK code.